raiden/network/proxies/token_network.py

```diff
@@ -955,94 +1323,70 @@ def settle(
+                error_prefix = 'settle call failed'
                 gas_limit = safe_gas_limit(gas_limit, GAS_REQUIRED_FOR_SETTLE_CHANNEL)
 
                 transaction_hash = self.proxy.transact(
                     'settleChannel',
                     gas_limit,
                     channel_identifier,
-                    partner,
-                    partner_transferred_amount,
-                    partner_locked_amount,
-                    partner_locksroot,
-                    self.node_address,
-                    transferred_amount,
-                    locked_amount,
-                    locksroot,
-                )
-            else:
-                gas_limit = self.proxy.estimate_gas(
-                    'settleChannel',
-                    channel_identifier,
-                    self.node_address,
-                    transferred_amount,
-                    locked_amount,
-                    locksroot,
-                    partner,
-                    partner_transferred_amount,
-                    partner_locked_amount,
-                    partner_locksroot,
+                    *args,
                 )
-                gas_limit = safe_gas_limit(gas_limit, GAS_REQUIRED_FOR_SETTLE_CHANNEL)
+                self.client.poll(transaction_hash)
+                receipt_or_none = check_transaction_threw(self.client, transaction_hash)
 
-                transaction_hash = self.proxy.transact(
-                    'settleChannel',
-                    gas_limit,
-                    channel_identifier,
-                    self.node_address,
-                    transferred_amount,
-                    locked_amount,
-                    locksroot,
-                    partner,
-                    partner_transferred_amount,
-                    partner_locked_amount,
-                    partner_locksroot,
-                )
+        transaction_executed = gas_limit is not None
+        if not transaction_executed or receipt_or_none:
+            if transaction_executed:
+                block = receipt_or_none['blockNumber']
+            else:
+                block = 'pending'
 
-            self.client.poll(transaction_hash)
-            receipt_or_none = check_transaction_threw(self.client, transaction_hash)
-            if receipt_or_none:
-                log.critical('settle failed', **log_details)
-                self._check_channel_state_for_settle(
-                    self.node_address,
-                    partner,
-                    channel_identifier,
-                )
-                raise TransactionThrew('Settle', receipt_or_none)
+            self.proxy.jsonrpc_client.check_for_insufficient_eth(
+                transaction_name='settleChannel',
+                transaction_executed=transaction_executed,
+                required_gas=GAS_REQUIRED_FOR_SETTLE_CHANNEL,
+                block_identifier=block,
+            )
+            msg = self._check_channel_state_after_settle(
```

Comment:
why are all of these functions returning to raise? is this really necessary?

Reply:
> why are all of these functions returning to raise?

To allow for some flexibility in how to handle the result of the check. For example in some case of `_check_channel_state_for_close` a warning is logged and in some not.